sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetCachedBatchSerializer.scala

```diff
@@ -690,162 +861,25 @@ protected class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer {
       origRequestedSchema: Seq[Attribute],
       sharedConf: Broadcast[Map[String, String]]) {
 
-    val conf: SQLConf = getConfFromMap(sharedConf)
-    val hadoopConf: Configuration = getHadoopConf(origRequestedSchema.toStructType, conf)
-    val options: ParquetReadOptions = HadoopReadOptions.builder(hadoopConf).build()
-
-    def getInternalRowIterator: Iterator[InternalRow] = {
-
-      /**
-       * This iterator converts an iterator[CachedBatch] to an iterator[InternalRow].
-       *
-       * This makes it unlike a regular iterator because CachedBatch => InternalRow* is a 1-n
-       * relation. The way we have implemented this is to first go through the
-       * iterator[CachedBatch] (cbIter) to look for a valid iterator (iter) i.e. hasNext() => true.
-       * Then every time next() is called we return a single InternalRow from iter. When
-       * iter.hasNext() => false, we find the next valid iterator in cbIter and the process
-       * continues as above.
-       */
-      new Iterator[InternalRow]() {
```

Comment:
I love how this complexity is just hidden away in the new iterator by returning an Iterator[InternalRow]